config.go

```diff
@@ -58,12 +58,17 @@ const (
 	AgentAuth     AuthMethod = "agent"
 )
 
-func LoadConfig(filepath string, isDefault bool, cfg *Config) error {
-	f, err := os.Open(filepath)
+func LoadConfig(filePath string, isDefault bool, cfg *Config) error {
+	f, err := os.Open(filePath)
 	if err != nil {
 		if os.IsNotExist(err) {
 			if isDefault {
-				f, err = os.Create(filepath)
+				dir := filepath.Dir(filePath)
+				if err := os.MkdirAll(dir, os.ModePerm); err != nil {
```

Comment:
`0755` is the more typical set of permissions for a user directory from my experience.

Reply:
You got it.